src/App.tsx

```diff
@@ -1,26 +1,275 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+/* eslint-disable @typescript-eslint/no-unused-vars */
 
-const USER_ID = 0;
+import React, { useEffect, useState, useRef, useCallback } from 'react';
+import * as todosService from './api/todos';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+// eslint-disable-next-line import/extensions
+import { Header } from './components/Header';
+// eslint-disable-next-line import/extensions
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState('');
+  const [filter, setFilter] = useState(Status.All);
+  const [activeTodosCount, setActiveTodosCount] = useState(0);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+    }, 0);
+
+    return () => clearTimeout(timer);
+  }, []);
+
+  const focusInput = useCallback(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, []);
+
+  useEffect(() => {
+    setLoading(true);
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setError('Unable to load todos'))
+      .finally(() => {
+        setLoading(false);
+        if (query !== '') {
+          focusInput();
+        }
+      });
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
  setLoading(true);
  todosService
    .getTodos()
    .then(setTodos)
    .catch(() => setError('Unable to load todos'))
    .finally(() => setLoading(false));
}, []);

useEffect(() => {
  if (query !== '') {
    focusInput();
  }
}, [query, focusInput]);
```

Consider splitting this `useEffect` to 2 separate hook calls - each of them makes different things (1st is for todos fetching, 2nd - just about focusing input)